crates/bevy_pbr/src/render/morph.rs

```diff
@@ -80,25 +107,32 @@ fn add_to_alignment<T: NoUninit + Default>(buffer: &mut RawBufferVec<T>) {
 // Notes on implementation: see comment on top of the extract_skins system in skin module.
 // This works similarly, but for `f32` instead of `Mat4`
 pub fn extract_morphs(
-    mut morph_indices: ResMut<MorphIndices>,
-    mut uniform: ResMut<MorphUniform>,
+    morph_indices: ResMut<MorphIndices>,
+    uniform: ResMut<MorphUniforms>,
     query: Extract<Query<(Entity, &ViewVisibility, &MeshMorphWeights)>>,
 ) {
-    morph_indices.clear();
-    uniform.buffer.clear();
+    // Borrow check workaround.
+    let (morph_indices, uniform) = (morph_indices.into_inner(), uniform.into_inner());
+
+    // Swap buffers. We need to keep the previous frame's buffer around for the
```

Comment:
I think your way is a bit cleaner, but this is how I did it in my PR https://github.com/JMS55/bevy/blob/3334d0bd5ff643f54a605672a70d05b6b271b455/crates/bevy_pbr/src/render/morph.rs#L93-L110. Can't remember why they look different off the top of my head.